base_geoengine/geo_db.py

```diff
@@ -18,22 +18,36 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
+"""Helper to setup Postgis"""
+from openerp.exceptions import MissingError
+from openerp.tools import ustr
 
-from osv import fields, osv
+import logging
 
-from base_geoengine import geo_model
+logger = logging.getLogger('geoengine.sql')
 
 
-class SaleOrder(geo_model.GeoModel):
-    """Add geo_point to sale.order"""
-    _name = "sale.order"
-    _inherit = "sale.order"
-    _columns = {'geo_point' : fields.geo_related('partner_invoice_id',
-                                                 'geo_point',
-                                                 string='Sale order coordinate',
-                                                 type='geo_point',
-                                                 relation='res.partner.address',
-                                                 dim=2, # mandatory arg
-                                                 srid=900913)}# mandatory arg
-SaleOrder()
-
+def init_postgis(cr):
+    """ Initialize postgis
+    """
+    cr.execute("""
+        SELECT
+            tablename
+        FROM
+            pg_tables
+        WHERE
+            tablename='spatial_ref_sys';
+    """)
+    check = cr.fetchone()
+    if check:
+        return {}
+    try:
+        cr.execute("""
+        CREATE EXTENSION postgis;
+        CREATE EXTENSION postgis_topology;
+    """)
+    except Exception as exc:
+        raise MissingError(
+            'Error, Can not initialize spatial postgis function.'
+            ' Database user may have to be superuser and postgres/postgis '
+            'extentions and dev header have to be installed', ustr(exc))
```

Comment:
Notpicking, but for consistence, add the spaces at the end of the lines. Also add a period at the end.